File: TypeScript/tests/reverse_integer.test.ts
import { assertEquals } from "@std/assert";
import { reverse } from "../src/reverse_integer.ts";

const cases = [
  { x: 123, want: 321 },
  { x: -123, want: -321 },
  { x: 120, want: 21 },
  { x: 1534236469, want: 0 },
  { x: -2147483648, want: 0 },
  { x: 900000, want: 9 },
];

for (let i = 0; i < cases.length; i++) {
  const t = cases[i];
  Deno.test(`reverseInteger - test case ${i + 1}`, () => {
    const result = reverse(t.x);
    assertEquals(result, t.want);
  });
}

File: C++/tests/reverse_integer_test.cpp
#include <gtest/gtest.h>
#include <reverse_integer.hpp>

class ReverseIntegerTest : public ::testing::Test {
protected:
  Solution solution;
};

TEST_F(ReverseIntegerTest, reverse_positive_integer) {
  int x = 123;
  int want = 321;

  int actual = solution.reverse(x);
  EXPECT_EQ(actual, want);
}

TEST_F(ReverseIntegerTest, reverse_negative_integer) {
  int x = -123;
  int want = -321;

  int actual = solution.reverse(x);
  EXPECT_EQ(actual, want);
}

TEST_F(ReverseIntegerTest, reverse_integer_with_trailing_zeros) {
  int x = 120;
  int want = 21;

  int actual = solution.reverse(x);
  EXPECT_EQ(actual, want);
}

TEST_F(ReverseIntegerTest, encounter_overflow) {
  int x1 = -2147483648;
  int want1 = 0;

  int actual1 = solution.reverse(x1);
  EXPECT_EQ(actual1, want1);

  int x2 = 2147483647;
  int want2 = 0;

  int actual2 = solution.reverse(x2);
  EXPECT_EQ(actual2, want2);
}